### ISSUE TITLE
Locale-related fonts has many platform dependencies

### ISSUE THREAD

**Issue by hibiyasleep** [OpenRCT2/OpenRCT2#3000]
**OS:** OS X, Windows without MS PGothic or any system without required font
**Version:** 0.0.4
**Commit:** 7d158b9, current latest master

Explanation of the issue...
- [v] Original bug?

**Steps to reproduce:**
1. Delete or disable(OS X) gulim.ttc
2. Game Options > Options > Language > Korean.
3. ta-da

---

I now trying to play RCT2 on mac.
I changed my conf to japanese/korean, and allmost all character is displayed in many squares.
So I dived into code, and I found that font [configuration is hardcoded](https://github.com/OpenRCT2/OpenRCT2/blob/876c4d369aa86b819cb022995aaa18f06db75e63/src/localisation/language.cpp#L50).
MS PGothic does not installed on OS X, and Gulim is `Comic Sans of Korean` so I disabled it.
Anyway, in short, It cannot be made configurable? or at least, can't be specified fallback unicode fonts?

**Comment by janisozaur** [comment]
On Linux we use [FontConfig](https://www.freedesktop.org/wiki/Software/fontconfig/), a system designed to handle such issues in a system-wide manner, so I consider Linux much less affected than the other platforms, though I see your point.

Also, 7d158b9 is not `master`, but `develop`.

One thing that is a bit problematic is that the font rendering is done completely within the indexed (paletted) image, so not all fonts look properly when used this way. I believe the fonts were chosen by users of languages they belong to as ones that render properly (cf. #2317, #2336). I have once tried replacing default Arial with something more pixel-art-ish, but all of the fonts I tried have had issues in various places, so it's not exactly easy choosing a proper font (though certainly possible).

Just providing some context.

**Comment by IntelOrca** [comment]
See #2269, for the feature request to specify the font it uses.

**Comment by marijnvdwerf** [comment]
I would expect the font selection to be done using font stacks instead of specific fonts. For OS X support, it should probably contain 'Arial Unicode MS', as that one renders fairly acceptable.

**Comment by AaronVanGeffen** [comment]
We have since added support for font families, making use of fonts that are present in vanilla macOS.
